source/game_sa/Entity/Ped/CPed.cpp

```diff
@@ -502,7 +503,151 @@ void CPed::UpdatePosition()
 // Converted from thiscall void CPed::ProcessBuoyancy(void) 0x5E1FA0
 void CPed::ProcessBuoyancy()
 {
+#ifdef USE_DEFAULT_FUNCTIONS
     ((void(__thiscall *)(CPed*))0x5E1FA0)(this);
+#else
+    if (bInVehicle)
+        return;
+
+    float fBuoyancyMult = 1.1F;
+    if (m_nPedState == ePedState::PEDSTATE_DEAD || m_nPedState == ePedState::PEDSTATE_DIE)
+        fBuoyancyMult = 1.8F;
+
+    float fBuoyancy = fBuoyancyMult * m_fMass / 125.0F;
+    CVector vecTurnSpeed;
+    CVector vecBuoyancy;
+    if (!mod_Buoyancy.ProcessBuoyancy(this, fBuoyancy, &vecTurnSpeed, &vecBuoyancy)) {
+        physicalFlags.bTouchingWater = false;
+        auto pSwimTask = m_pIntelligence->GetTaskSwim();
+        if (pSwimTask)
+            pSwimTask->m_fSwimStopTime = 1000.0F;
+
+        return;
+    }
+
+    if (bIsStanding) {
+        auto& pStandingOnEntity = m_pContactEntity;
+        if (pStandingOnEntity && pStandingOnEntity->m_nType == eEntityType::ENTITY_TYPE_VEHICLE) {
+            auto pStandingOnVehicle = reinterpret_cast<CVehicle*>(pStandingOnEntity);
+            if (pStandingOnVehicle->m_nVehicleClass == eVehicleClass::CLASS_BIG
+                && !pStandingOnVehicle->physicalFlags.bDestroyed) {
+
+                physicalFlags.bSubmergedInWater = false;
+                auto pSwimTask = m_pIntelligence->GetTaskSwim();
+                if (!pSwimTask)
+                    return;
+
+                pSwimTask->m_fSwimStopTime += CTimer::ms_fTimeStep;
+                return;
+            }
+        }
+    }
+
+    if (m_pPlayerData) {
+        const auto& vecPedPos = GetPosition();
+        float fCheckZ = vecPedPos.z - 3.0F;
+        CColPoint lineColPoint;
+        CEntity* pColEntity;
+        if (CWorld::ProcessVerticalLine(vecPedPos, fCheckZ, lineColPoint, pColEntity, false, true, false, false, false, false, nullptr)) {
+            if (pColEntity->m_nType == eEntityType::ENTITY_TYPE_VEHICLE) {
+                auto pColVehicle = reinterpret_cast<CVehicle*>(pColEntity);
+                if (pColVehicle->m_nVehicleClass == eVehicleClass::CLASS_BIG
+                    && !pColVehicle->physicalFlags.bDestroyed
+                    && pColVehicle->GetMatrix()->GetUp().z > 0.0F) {
+
+                    physicalFlags.bSubmergedInWater = false;
+                    return;
+                }
+            }
+        }
+    }
+
+    // Those 4 are called for some reason
+    CTimeCycle::GetAmbientRed();
+    CTimeCycle::GetAmbientGreen();
+    CTimeCycle::GetAmbientBlue();
+    rand();
```

Comment:
```suggestion
    // Those 4 are called for some reason
    /*
    CTimeCycle::GetAmbientRed();
    CTimeCycle::GetAmbientGreen();
    CTimeCycle::GetAmbientBlue();
    rand();
    */
```
I usually comment unused code.